### PR TITLE
Add OKRs for 2019.

### PR DESCRIPTION
**Pull request by teamdandelion**
Add SourceCred 2019 OKRs

Thanks to @anthrocypher, @momack2, @wchargin, and @sternhenri  for helping produce
this document.

Test plan: Click every link, check that tables are formatted correctly,
review for typos etc.

### CODE REVIEW

File: okrs_2019.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+# SourceCred 2019 OKRs
+
+This document contains SourceCred's objectives, initiatives, and metrics for
+2019, generated as part of Protocol Lab's annual planning process.
```
Comment (wchargin) [inline okrs_2019.md]: > Protocol Lab's

sp.: Protocol Labs’s

File: okrs_2019.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+us in to a particular set of initiatives--we know that as time goes on, new
+opportunities will present themselves, and some of the existing initiatives
+will prove ill-founded or infeasible. Also, our metrics have known deficiencies
+(e.g. we don't yet know how to measure the quality of a cred attribution).
+
+As such, this is intended to be a living document. We will update it both to
+record progress on our goals, and to modify and retarget our goals as we gain
+more information. Since this document is in Git, all revisions will be
+publically visible. If we remove or replace a goal, initiave, or metric, we'll
+clearly explain the rationale for doing so in the associated commit.
```
Comment (wchargin) [inline okrs_2019.md:22]: Cool.

File: okrs_2019.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+
+# SourceCred Goals
+
+## Improve Attribution Quality
+
+SourceCred's core function is to produce cred attributions for open-source
```
Comment (wchargin) [inline okrs_2019.md:28]: > Improving the quality of those attributions is the our highest
> priority

It is good to note this explicitly. I hope that it continues to remain
the top priority.

File: okrs_2019.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+- improving test coverage
+- maintaining the build
+- triaging issues
+- helping to resolve interpersonal conflicts
+- planning for future development
+- refactoring difficult code
+- paying down technical debt
+- making the project fun to work on
+
+### Initiatives
```
Comment (wchargin) [inline okrs_2019.md:49]: Missing here: Add identity resolution (SourceCred should understand that
my GitHub account, my various Git authorships, and my IRC handle all
represent the same entity, and should interpret cred relationships
accordingly).

Comment (wchargin) [inline okrs_2019.md:49]: Missing here: Define SourceCred’s approach to gaming resistance: specify
which kinds of attacks are considered “in scope”, develop mechanisms to
counteract these attacks, and assess the effectiveness of the
prophylaxis.

Comment (wchargin) [inline okrs_2019.md:49]: I want to call out here that a number of these can be done, to put it
bluntly, either “hastily” or “thoughtfully”. To give a few examples
(most of which we’ve talked about offline to some degree):

  - In terms of code analysis: What does it mean for “files” to be
    tracked by SourceCred? What is a “file”, given the context of a
    commit history? How are (say) functions identified across files and
    across time? What happens when someone refactors a function? What
    happens at merge commits? How can we make our analyses
    computationally tractable (do we need something like Hercules?)?

  - In terms of time-based cred analysis: What exactly do time-weights
    apply to (nodes? edges? something else?)? How does time-weighting
    interact with the inherent transitivity of the graph? If an
    unappreciated contribution from 2010 is made much more valuable by a
    contribution from 2015, how does the score for that 2010
    contribution change as a function of the weighting? For what
    purposes is time-weighted cred intended to be used?

  - In terms of heuristics: How do heuristics compose? When multiple
    heuristics apply to an item, with either contradictory or
    reinforcing assessments, how do we compile those evaluations? How do
    heuristics interact with “pass-through” semantics (a judgment that
    “this code reformatting commit is unimportant” may indicate to
    reduce the commit’s overall influence on the graph, whereas a
    judgment that “this merge commit is unimportant” more indicates that
    the merge commit’s incoming cred is still valuable but should be
    funneled through to its parents rather than its authors)?

  - In terms of “cred decomposition”: Should this satisfy some kind of
    additive properties, like “the cred to Node X is equal to the sum of
    (cred to Node X due to nodes in Part P) across Parts P that
    partition the space? Is there some natural interpretation of
    weighted cred decomposition? What does it mean to
    isolate/scope/decompose cred in such a cyclic structure?

Each of the above should have a design doc that specifies what
objectives were considered, what objectives were out of scope, what
implementations were considered, the relative advantages and
disadvantages of the implementations, and why the final decision was so
made. These are critical parts of SourceCred; you can’t just do them on
the fly.

File: okrs_2019.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  - * [ ] Document the plugin system, and how to add a new one
+  - * [ ] Add API docs
+- * [ ] Produce engaging blog posts
+- * [ ] Produce engaging audiovisual content
+
+### Metrics
+
+| Metric                                  | Goal | Q4 | Q1 | Q2 | Q3 | Q4 |
+| --------------------------------------- | ---- | -- | -- | -- | -- | -- |
+| # of contributors w/ >10 cred           | 15   | 2  |    |    |    |    |
```
Comment (wchargin) [inline okrs_2019.md]: > \# of contributors w/ >10 cred

I suspect that this is infeasible. Git itself is by any means a
successful open source project, with regular ICs both as independents
and as employed by leading tech companies, and it has merely 11
contributors with ≥10 cred. (For comparison, Git has 479 contributors
each of whom has at least 5 commits, and has 74 contributors each of
whom has at least 74 commits.)

File: okrs_2019.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+| # of unique code committers             | 30   | 3  |    |    |    |    |
+| # of core team members                  | 3    | 1  |    |    |    |    |
+| # of Discord community members          | 500  | 61 |    |    |    |    |
```
Comment (wchargin) [inline okrs_2019.md:142]: > \# of Discord community members

This is something of a strange goal when there is really very little
incentive for people to join our Discord. The pattern that we’ve seen is
that interested parties join and then lurk indefinitely—are you trying
to get 450 more lurkers?

Perhaps a more insightful metric in this space would be something like
“per-month average number of distinct people who discuss salient aspects
of SourceCred”. (So, thoughts about cred-gaming, attributions,
deployments, implementation are all fair game; scheduling a video call
is not.)

File: okrs_2019.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+On the initiative side, we want to improve the documentation around all parts of
+SourceCred so that the project is easier to get up to speed on. We also intend
+to produce content explaining SourceCred, like blog posts and video talks.
+
+### Initiatives
+- * [ ] Improve SourceCred's documentation
+  - * [ ] Document the overall vision, mission, and philosophy
+  - * [ ] Document the overall technical architecture
+  - * [ ] Document the attribution algorithm (e.g. by writing a whitepaper)
```
Comment (wchargin) [inline okrs_2019.md]: > Document the attribution algorithm (e.g. by writing a whitepaper)

Suggest deprioritizing unless and until we have something interesting.
What we have now does not merit a white paper; it merits a paragraph in
our general documentation. There’s nothing special about it.

File: okrs_2019.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+  - * [ ] Document the plugin system, and how to add a new one
+  - * [ ] Add API docs
```
Comment (wchargin) [inline okrs_2019.md]: > Add API docs

This does not make sense: we do not have an API. Perhaps “Design API” is
the real action item—but that’s not (just) a documentation question.

File: okrs_2019.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+## SourceCred Adoption
+
+Right now, SourceCred is a prototype in search of some real users. Actual users
+will be a great source of feedback on the quality of our attributions, and also
+will bring more people into our community.
+
+### Initiatives
+
+One initiative we’re quite excited about is a planned integration with
+[OpenCollective]. We intend to embed SourceCred results into every open-source
```
Comment (wchargin) [inline okrs_2019.md]: > OpenCollective

sp.: Open Collective (throughout)

File: okrs_2019.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+collective’s page (e.g. here is the [page for Webpack]). This way, collectives
+can thank their leading contributors as well as their financial sponsors. We
+have enthusiastic buy-in from the OpenCollective team, and think this could be
+the start of a great partnership, in addition to bringing more attention and
+users to SourceCred.
+
+[OpenCollective]: https://opencollective.com/
+[page for Webpack]: https://opencollective.com/webpack#contributors
+
+- [ ] Integrate with OpenCollective ([#935])
```
Comment (wchargin) [inline okrs_2019.md]: > Integrate with OpenCollective

Not listed as a requirement: some hosting on our part. Is it expected
that Open Collective will be managing all of this (maintaining
SourceCred instances, keeping them up to date, serving recent
attributions, …?)

Not listed as a requirement: better stability in the Mirror module.
Hardcoded blacklisting won’t cut it. Probably our support for deletions
won’t cut it either. As a quick fix we can always include some logic to
throw away the cache if the Mirror module gives one of these errors, but
that won’t be fun for (a) whoever pays for hosting and (b) anyone who
wants to see quickly updated attributions.

File: okrs_2019.md [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+downstream projects. For now, we'll measure "engagment", which is defined as:
+the project has taken a public action to use SourceCred, e.g. by linking to the
+cred attribution in their README, adding cred configuration files to their repository,
+or integrating a cred bot on GitHub. Actions that can happen without direct engagement
+from the project (e.g. listing their cred on OpenCollective) don't count.
+
+For the purposes of the metrics below, "medium-sized projects" have at least 30 users
+in their cred attribution, and "large-sized projects" have at least 100.
+
+We'd like to see crypto projects start to use SourceCred as a way to transparently
```
Comment (wchargin) [inline okrs_2019.md:201]: > We'd like to see crypto projects start to use SourceCred as a way to
> transparently distribute tokens to contributors. For purposes of the
> metric, we're tracking public announcements or active distributions by
> well-known projects (i.e. no unknown ICOs).

I have some concerns about the incentives that this creates. Let me try
to express them clearly and get back to you.

**Review by wchargin — COMMENTED** [review]
I’m happy to see this document. Thank you for writing it. Here are my
initial thoughts.

File: okrs_2019.md (hunk 4 above)
Comment (teamdandelion) [reply okrs_2019.md:49]: Done

Comment (teamdandelion) [reply okrs_2019.md:49]: Agreed on all counts. I re-wrote this section of the document to try to incorporate your feedback, ptal.

File: okrs_2019.md (hunk 5 above)
Comment (teamdandelion) [reply okrs_2019.md]: Good point. I've removed it pending us coming up with something that we have confidence in.

File: okrs_2019.md (hunk 6 above)
Comment (teamdandelion) [reply okrs_2019.md:142]: Yeah, those would be better metrics. The # of discord community members has the advantage that it's trivial to measure, while the other suggestions are not. You can think of it as kind of like counting GitHub stars, it indicates that someone was interested enough to engage with SourceCred at least once.

I'm going to leave this in, while acknowledging that it's far from a perfect metric. If we have an (easily implemented) better metric I'd be happy to swap it.

File: okrs_2019.md (hunk 7 above)
Comment (teamdandelion) [reply okrs_2019.md]: Sure thing. I amended the "improve cred attribution" to more clearly express that many of the initiatives are in fact research problems, and we should start by clearly documenting those problems. That is actually what I was getting at here.

File: okrs_2019.md (hunk 10 above)
Comment (teamdandelion) [reply okrs_2019.md]: Yeah, we'll need to figure out hosting. I think initially a cron style job on AWS that pushes json data onto GitHub pages will suffice.

In the spirit of "this is a living document", I am going to spec out the work for this once we're to implementing it.

File: okrs_2019.md (hunk 11 above)
Comment (teamdandelion) [reply okrs_2019.md:201]: Ack.

File: okrs_2019.md (hunk 4 above)
Comment (teamdandelion) [reply okrs_2019.md:49]: Added.

File: okrs_2019.md [hunk 12]
```diff
@@ -0,0 +1,229 @@
+# SourceCred 2019 OKRs
+
+This document contains SourceCred's objectives, initiatives, and metrics for
+2019, generated as part of Protocol Labs's annual planning process.
+For an overview of SourceCred itself, see [the overview](overview.md).
+
+The purpose of this document is to provide clarity around what the project's
+goals and priorities are, and accountability to see if we are making effective
+progress towards achieving those goals. However, this is not intended to lock
```
Comment (sternhenri) [inline okrs_2019.md]: I might address the reader more directly here with something along the lines of:

> We hope this enables you to jump into SourceCred with a clear sense of our goals for 2019, and how our granular work on a regular basis translates to the larger vision for the project.

File: okrs_2019.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+## Improve Attribution Quality
+
+SourceCred's core function is to produce cred attributions for open-source
+projects. Improving the quality of those attributions is the our highest
+priority. We want SourceCred to do a great job of assigning cred both to
+contributors, and also to specific contributions.
+
+SourceCred's current implementation is a promising start, but still leaves much
+to be desired. It does a decent job of assigning cred to contributors; however,
+it struggles with the greater challenge of valuing specific contributions.
```
Comment (sternhenri) [inline okrs_2019.md]: Is there a succinct way of showing off the data that enables you to make those twin claims (fair job of contributor cred, bad job of contribution cred)? Straw man is "Based on early product testing"

File: okrs_2019.md [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+[sourcecred/notes#1]: https://github.com/sourcecred/notes/issues/1
+
+### Metrics
+
+Coming up with metrics for attribution quality is challenging, since there
+isn't a readily available ground truth to compare against. Right now, the best
+metric we have is asking the contributors to a project to rate the quality of
+our attribution. The "cred rating" in the table below is the average rating
+that contributors to a project give its cred attribution, measured on a
+10-point scale.
```
Comment (sternhenri) [inline okrs_2019.md:99]: Would love a bit more on this. On two levels:

1- Meta point: will you stick with "asking contributors" as a best-in-class metric for 2019? You seem to imply it's the least worst, which invites the question "what will be the upgrade" but don't elaborate. A quick "we will keep working off of this metric as ground truth in 2019 but include this section as a reminder of the importance future developments in this area of study will have on SourceCred" would help.

2 - Would love a link to a quick doc on the 10-point scale. This seems like it would be a highly highly subjective system, not just on a per-process basis, but based on a contributors' understanding of SourceCred and the underlying project itself (not to mention interpersonal dynamics, politics etc). Are you averaging scores from all contributors to get the "project" score? Are all contributors learning how to rate from the same prompt? Would love to understand better (not in scope for this doc)

3- Why do you have different goals for SC across both projects? Wouldn't you expect it to perform as well on both?

File: okrs_2019.md [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+SourceCred so that the project is easier to get up to speed on. We also intend
+to produce content explaining SourceCred, like blog posts and video talks.
+
+### Initiatives
+- * [ ] Improve SourceCred's documentation
+  - * [ ] Document the overall vision, mission, and philosophy
+  - * [ ] Document the overall technical architecture
+  - * [ ] Document the plugin system, and how to add a new one
+- * [ ] Produce engaging blog posts
+- * [ ] Produce engaging audiovisual content
```
Comment (sternhenri) [inline okrs_2019.md]: Specific numbers in the above "produce" will make them easier to gauge as KRs. eg "Produce 2 engaging blog posts quarterly"

File: okrs_2019.md (hunk 6 above)
Comment (sternhenri) [reply okrs_2019.md:142]: +1. I run into the issue of what does "salient" mean otherwise. Acknowledging the flaw in the metric with confidence it is directionally correct should do for now ;)

File: okrs_2019.md [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+
+These metrics mostly record the number of people who have engaged with the community,
+rather than its health and vibrancy. As such, they can be used as a general directional
+proxy, but shouldn't be taken too seriously. The good news is, if the community is
+thriving, it should be obvious. :)
+
+## SourceCred Adoption
+
+Right now, SourceCred is a prototype in search of some real users. Actual users
+will be a great source of feedback on the quality of our attributions, and also
```
Comment (sternhenri) [inline okrs_2019.md]: nit: I might strengthen the language here: actual users will be "the only real" source of feedback. Everything else is conjecture. (though users's opinions should be read carefully)

File: okrs_2019.md [hunk 17]
```diff
@@ -0,0 +1,229 @@
...
+We’re quite excited a planned integration with [Open Collective]. We intend to
+embed SourceCred results into every open-source collective’s page (e.g. here is
+the [page for Webpack]). This way, collectives can thank their leading
+contributors as well as their financial sponsors. We have enthusiastic buy-in
+from the Open Collective team, and think this could be the start of a great
+partnership, in addition to bringing more attention and users to SourceCred.
+
+To do this, we need to make a number of engineering improvements to SourceCred.
+The mirror module (which loads data from GitHub) needs to be made more robust
+to errors in GitHub's GraphQL implementation ([#996]). We also need a proper
```
Comment (sternhenri) [inline okrs_2019.md:166]: 996 broken link

File: okrs_2019.md (hunk 11 above)
Comment (sternhenri) [reply okrs_2019.md:201]: On a perhaps different topic, I don't love this measure of "engagement." I understand the trade-offs at hand (I think), but it creates an incentive for vocal participants over real participation. Both are key (the former for visibility and users) but I think engagement must include the latter, as the sole measure of SourceCred's actual usefulness to projects (vs whether it sounds good to people, or helps them sell).

Potential solutions:
  - (1) A quick callback to SC from downstream projects that helps them automate tracking of SC use: how often they integrate it to a page, how often that page is viewed, etc (super centralized and pretty bad)
 - (2) A more qualitative metric such as a measure of engagement derived from a standard questionnaire SC provides users on a quarterly basis (participation rate there is a good measure onto itself)
- (3) Some tracking of discussions surrounding specific downstream work in the SC forums themselves (hard to measure)
- (4) Simply an acknowledgement that this is a flawed metric and will be upgraded once early projects are onboarded.

I would opt for 2 or 4 to start

File: okrs_2019.md [hunk 18]
```diff
@@ -0,0 +1,229 @@
...
+### Initiatives
+
+Making SourceCred useful means finding a way to make the cred results salient
+to contributors to a project. Time-weighted cred (as described elsewhere in
+this document) will help a lot, since filtering cred to recent contributions
+will make the cred results more timely and relevant, especially to new
+contributors who won't quickly show up on the full-history analysis.
+
+We also may want to explore things like browser extensions or GitHub bots that
+surface cred information more readily.
```
Comment (sternhenri) [inline okrs_2019.md]: This Objective is a bit unclear to me at this point. 

Can you rephrase after this, perhaps you mean:

"Evaluate and document 5 potential initiatives to drive SC usage and implement 2 of them"

**Review by sternhenri — COMMENTED** [review]
I left some thoughts in the doc itself but my high level would be that I like the structure and overall directions for the project. I might personally emphasize usage and engagement over the system's quality at this juncture (build something users want), which could mean flipping the order on these, but this is personal preference.

I have two main points:
1) Right now, these OKRs read a bit more like a statement of purpose and list of high-level objectives for the year. What I like about OKRs is how actionable they should feel, and easy to gauge they should be. 

I would recommend a few formatting changes in that direction, such as:
- Number each Objective for SC under each of the sections, to end with a list of probably 4-6 objectives under the 4 main areas you mention here
- Number each KR for each objective, rewritten in a measurable way: 
  - eg "Add identity resolution" becomes "SC has identity resolution," or "Produce engaging blog posts" -> becomes "SC has produced 6 engaging posts": there should be no doubt as to whether the result is there at year's end.

Specifically, a KR from this objective being met might be: this OKR sheet can be taken and reduced to a list of bulleted Os and KRs that to measure impact of ongoing work on a quarterly basis. I will add that this document in this format remains extraordinarily useful, not least as a public rationale for the set of KRs you've chosen!

2) This brings me to quarterly work: I think splitting out Q1 KRs from these 2019 objectives would be useful (this could be as helpful as just putting a "Q1" tag next to relevant ones). I fear that having a series of ambitious yearly objectives which you measure throughout the year means a risk of having some few complete items at year's end. Instead by moving things to be fully complete sequentially (rather than in parallel) on a quarterly basis, you can move your time/energy more effectively (with no possible excuses around timing).

File: okrs_2019.md [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+integration is one.
+
+| Metric                                  | Goal | Q4 | Q1 | Q2 | Q3 | Q4 |
+| --------------------------------------- | ---- | -- | -- | -- | -- | -- |
+| # of engaged medium-sized projects      | 50   | 0  |    |    |    |    |
+| # of engaged larged-sized projects      | 5    | 0  |    |    |    |    |
+| # of announced token distributions      | 3    | 0  |    |    |    |    |
+| # of 3rd party integrations             | 3    | 0  |    |    |    |    |
+
+## SourceCred Helps Projects
```
Comment (sternhenri) [inline okrs_2019.md]: The set of KRs below is not actionable enough yet imo. May want to dwindle down to the very specifics (at the risk of picking the wrong ones for now)!

File: okrs_2019.md (hunk 15 above)
Comment (teamdandelion) [reply okrs_2019.md]: Done

File: okrs_2019.md (hunk 19 above)
Comment (teamdandelion) [reply okrs_2019.md]: This Objective is pretty far in the future; to engage with it, we need to solve a lot of issues like getting users, making cred engaging, adding heuristics so we can really start to try to change behaviors. By the time we're ready to start actually making progress here, we'll have a better idea of what to measure. I'm OK leaving this vague for now. #livingdocument

File: okrs_2019.md (hunk 17 above)
Comment (teamdandelion) [reply okrs_2019.md:166]: Fixed, thanks.

File: okrs_2019.md (hunk 16 above)
Comment (teamdandelion) [reply okrs_2019.md]: Done

File: okrs_2019.md (hunk 15 above)
Comment (teamdandelion) [reply okrs_2019.md]: Done.

File: okrs_2019.md (hunk 14 above)
Comment (teamdandelion) [reply okrs_2019.md:99]: > 1- Meta point: will you stick with "asking contributors" as a best-in-class metric for 2019? You seem to imply it's the least worst, which invites the question "what will be the upgrade" but don't elaborate. A quick "we will keep working off of this metric as ground truth in 2019 but include this section as a reminder of the importance future developments in this area of study will have on SourceCred" would help.

It would be great if we come up with a better metric during 2019, although doing so isn't a top priority right now. If we do come up with a better metric, we'll add it to the doc. (With historical measurements if possible for consistent tracking.) I expect we'll keep cred ratings around at least for all of the year so we can have consistent measurements.

> 2 - Would love a link to a quick doc on the 10-point scale. This seems like it would be a highly highly subjective system, not just on a per-process basis, but based on a contributors' understanding of SourceCred and the underlying project itself (not to mention interpersonal dynamics, politics etc). Are you averaging scores from all contributors to get the "project" score? Are all contributors learning how to rate from the same prompt? Would love to understand better (not in scope for this doc)

Good point. See https://github.com/sourcecred/mission/issues/9 where we can track figuring all of this out.

> 3- Why do you have different goals for SC across both projects? Wouldn't you expect it to perform as well on both?

Updated the doc to explain: 
> We have a higher goal for SourceCred's average rating for two reasons:
> - As a smaller and younger project, attributing its cred is simply easier
> - We think that having a highly engaged community will enable producing better
  attributions, and we expect the SourceCred community to be more engaged

File: okrs_2019.md (hunk 13 above)
Comment (teamdandelion) [reply okrs_2019.md]: It's based on early product testing :smile:.

File: okrs_2019.md (hunk 11 above)
Comment (teamdandelion) [reply okrs_2019.md:201]: I don't understand your concerns around engagement, or how it prioritizes vocal users. Here, a project is described as "engaged" if there is externally-visible evidence that the project has made a deliberate decision to use SourceCred. An example of what this might look like is, maybe as we add .credconfig files that allow projects to tune their weights and heuristics, and we count how many projects have merged these configs.

I do like the idea of coming up with better engagement metrics. The quarterly questionnaire is a nice idea. For now I'm inclined to leave this as it is, for me every metric in the doc is implicitly open to being upgraded.

File: okrs_2019.md (hunk 12 above)
Comment (teamdandelion) [reply okrs_2019.md]: Done

**Comment by teamdandelion** [comment]
Thank you for the detailed review feedback @sternhenri!

>  I might personally emphasize usage and engagement over the system's quality at this juncture (build something users want), which could mean flipping the order on these, but this is personal preference.

There are kinds of quality which can be easily improved after the system is widely deployed, and kinds of quality that are locked in by initial design decisions. In my mind, we haven't yet de-risked the initial design, which is why improving the quality is the top priority. An example is that if we were building a cryptonetwork, we would want to prove that it's secure, and that there's a viable path to scaling, before actually scaling it.

It's interesting to note that in the space of search engines, the space was resoundingly won by the group that had the best initial design, not by the groups that got engagement and users fastest. *(Note: I have not made a study of the competitive landscape of early search engines.)*

That said, getting users and adoption is clearly useful for us in improving attribution quality: more training data, more resources, more minds on the problem. Also, unlike with a cryptonetwork, updating SourceCred is relatively easy. So long as we stay with the core representation (aka the Contribution Graph) we can actually change the algorithm dramatically.

So... they're both goals that we'll work on, and I think putting quality first is a conservative and appropriate choice.

> Right now, these OKRs read a bit more like a statement of purpose and list of high-level objectives for the year. 

I'm fine with this; a statement of purpose, and prioritization of high-level objectives, seems to me to be the valuable part of the exercise. If this document proves cumbersome, e.g. because we need it to translate more cleanly into a sheet, we can update it at that time. (Also; pull requests accepted.)

> Splitting out Q1 OKRs

I agree, this would be really useful. Created https://github.com/sourcecred/mission/issues/10 to track.

File: okrs_2019.md (hunk 11 above)
Comment (sternhenri) [reply okrs_2019.md:201]: Fair enough, and I really like the .credconfig idea, which signals they've actually meaningfully engaged with SC. My issue with externally-visible signs as I initially read them (ie mention of SC in some public page for the project) is that it sets it a shallow bar as to what we consider an engaged user (in a way that the .credconfig file does less so). In time the hope is that using SC becomes a best practice in OSS development, and simply saying your project uses SC isn't good enough, but we're on the same page!

File: okrs_2019.md (hunk 14 above)
Comment (sternhenri) [reply okrs_2019.md:99]: This is great!

File: okrs_2019.md (hunk 19 above)
Comment (sternhenri) [reply okrs_2019.md]: Fair enough. May be worth adding a mention of this is a future objective or forward looking one...
